ui/app/AppLayouts/Onboarding2/OnboardingFlow.qml

```diff
@@ -157,7 +158,11 @@ SQUtils.QObject {
             onLoginRequested: (keyUid, method, data) => root.loginRequested(keyUid, method, data)
             onOnboardingCreateProfileFlowRequested: root.stackView.push(createProfilePage)
             onOnboardingLoginFlowRequested: root.stackView.push(loginPage)
-            onLostKeycard: root.stackView.push(keycardLostPage)
+            onLostKeycard: (keyUid) => {
+                               root.keyUidSubmitted(keyUid)
+                               root.stackView.push(keycardLostPage)
+                           }
```

Comment:
Basically there is no need to modify `LoginScreen` at all.
Here the handling could look like:

```suggestion
onLostKeycard: {
    root.keyUidSubmitted(loginScreen.selectedProfileKeyId)
    root.stackView.push(keycardLostPage)
}
```

Because `selectedProfileKeyId` is already exposed and can be used when any signal is emitted from `LoginScreen`.